plugins/h3/h3.cpp

```diff
@@ -0,0 +1,247 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2019 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "platform.h"
+#include "eclrtl.hpp"
+#include "h3.hpp"
+
+#include <vector>
+
+#define H3_VERSION "h3 plugin 1.0.0"
+ECL_H3_API bool getECLPluginDefinition(ECLPluginDefinitionBlock *pb)
+{
+    /*  Warning:    This function may be called without the plugin being loaded fully.
+     *              It should not make any library calls or assume that dependent modules
+     *              have been loaded or that it has been initialised.
+     *
+     *              Specifically:  "The system does not call DllMain for process and thread
+     *              initialization and termination.  Also, the system does not load
+     *              additional executable modules that are referenced by the specified module."
+     */
+
+    if (pb->size != sizeof(ECLPluginDefinitionBlock))
+        return false;
+
+    pb->magicVersion = PLUGIN_VERSION;
+    pb->version = H3_VERSION;
+    pb->moduleName = "lib_h3";
+    pb->ECL = NULL;
+    pb->flags = PLUGIN_IMPLICIT_MODULE;
+    pb->description = "ECL plugin library for uber h3\n";
+    return true;
+}
+
+#include <h3api.h>
+#define H3_INDEX_MINSTRLEN 17
+
+void toSetOf(bool &__isAllResult, size32_t &__lenResult, void *&__result, H3Index *buff, int maxBuffLen)
+{
+    int finger = 0;
+    for (int i = 0; i < maxBuffLen; ++i)
+    {
+        if (buff[i] == 0)
+            continue;
+
+        if (i > finger)
+        {
+            buff[finger] = buff[i];
+            buff[i] = 0;
```

Comment:
Probably not needed.

Reply:
I think it is, otherwise you would duplicate n+1 into n rather than moving it (and subsequent)

Reply:
I don't think so `buff[that Specific "I"]` never gets read again and will be overridden if/when finger gets there.

Reply:
I agree the line probably isn't needed

Reply:
So do I , I think